src/main/java/com/dawsonsystems/session/MongoManager.java

```diff
@@ -518,21 +444,33 @@ private void initDbConnection(String path) throws LifecycleException {
         addrs.add(new ServerAddress(host, getPort()));
       }
 
-      mongo = new MongoClient(addrs,
-          MongoClientOptions.builder()
+      Builder clientOptionsBuilder = MongoClientOptions.builder()
               .description("TomcatMongoSession[path=" + path + "]")
               .alwaysUseMBeans(true)
-              .connectionsPerHost(connectionsPerHost)
-              .build());
+              .connectionsPerHost(connectionsPerHost);
+
+      List<MongoCredential> mongoCredentials = new ArrayList<>();
+
+      if (sslTrustStorePem != null && sslTrustStorePem.length() > 0 && sslKeyStorePem != null && sslKeyStorePem.length() > 0) {
+        log.info("Detected Mongo SSL configuration: trust store PEM: " + sslTrustStorePem + ", key store PEM: " + sslKeyStorePem);
+        SSLContext sslContext = sslContextFromPEMs(new File(sslTrustStorePem), new File(sslKeyStorePem));
+        clientOptionsBuilder.sslEnabled(true);
+        clientOptionsBuilder.sslContext(sslContext);
+        mongoCredentials.add(MongoCredential.createMongoX509Credential());
+      } else {
+        log.info("Using an unencrypted connection to Mongo");
+      }
+
+      mongo = new MongoClient(addrs, mongoCredentials, clientOptionsBuilder.build());
 
       db = mongo.getDB(getDatabase());
       if (slaveOk) {
         db.setReadPreference(ReadPreference.secondaryPreferred());
       }
       db.setWriteConcern(WriteConcern.ACKNOWLEDGED);
       getCollection().createIndex(new BasicDBObject("lastmodified", 1));
-      log.info("Connected to Mongo " + host + "/" + database + " for session storage, slaveOk=" + slaveOk + ", " + (getMaxInactiveInterval() * 1000) + " session live time");
-    } catch (IOException e) {
+      log.info("Connected to Mongo " + host + "/" + database + " for session storage, slaveOk=" + slaveOk + ", " + (getSessionMaxAliveTime() * 1000) + " session live time");
+    } catch (RuntimeException | IOException e) {
       e.printStackTrace();
```

Comment:
Use logger to log severe message that includes stack trace, rather than calling `printStackTrace`.

Reply:
Made the change per suggestion, but is it possible the exception was thrown due to writing logs? In that case, the error could get lost.  BTW, the e.printStackTrace() code was from the original code we forked from.